src/routes/index.ts

```diff
@@ -1,7 +1,181 @@
+        req.on('data', (data) => {
+            csvString += data;
+        });
+        req.on('end', () => {
+            async function invite() {
+                try {
+                    // Parse the csvString
+                    const result = Papa.parse(csvString, {
+                        header: true,
+                    });
+                    const inviteeList = result.data as Array<InviteeData>; // Validate these fields on frontend
+                    const unsubSet = new Set(
+                        await Notifications.getUnsubList(region as string) // Checked if undefined earlier
+                    );
+                    const filteredInviteeList = inviteeList.filter(
+                        (item: InviteeData) => {
+                            return !unsubSet.has(item.email);
+                        }
+                    );
+                    if (filteredInviteeList.length === 0) {
+                        res.status(400).send('No valid invitees');
+                        // throw new ClientError('No valid invitees') // TODO FIX
+                    }
+                    const results = await Invite.inviteMany(
+                        filteredInviteeList,
+                        MoC as string, // Checked if undefined earlier
+                        topic as string,
+                        eventDateTime as string,
+                        constituentScope as string,
+                        deliveryTime
+                    );
+                    logger.print(JSON.stringify(results));
+                    res.status(200).send();
+                } catch (e) {
+                    logger.err(e);
+                }
+            }
+            // eslint-disable-next-line no-void
+            void invite();
+        });
+    } catch (e) {
+        logger.err(e);
+        next(e);
+    }
+});
+
+export interface SubscribeData {
+    email: string;
+    region: string;
+}
+
+router.post('/subscribe', async (req, res, next) => {
+    try {
+        const data = req.body as SubscribeData;
+        if (data.email === undefined || data.region === undefined) {
+            throw new ClientError('Invalid Data');
+        }
+        const isSubscribed = await Notifications.isSubscribed(
+            data.email,
+            data.region
+        );
+        if (isSubscribed) {
+            res.status(400).send('Already subscribed');
+            // throw new ClientError('Already subscribed.'); // TODO FIX
+        }
+        const isUnsubscribed = await Notifications.isUnsubscribed(
+            data.email,
+            data.region
+        );
+        // By default any subscriber will be removed from the unsub list
+        if (isUnsubscribed) {
+            await Notifications.removeFromUnsubList(data.email, data.region);
+            await Subscribe.mailgunDeleteFromUnsubList(data.email);
+        }
+        await Notifications.addToSubList(data.email, data.region);
+        res.status(200).send();
```

Comment:
I think it's possible that this code is `res.send()`ing twice. On line 130, if `isUnsubscribed` is true, then what's to stop line 131 and 144 from running in the same call?